metrics/ci_integrations/test/source/github_actions/adapter/github_actions_source_client_adapter_test.dart

```diff
@@ -1440,6 +1018,143 @@ void main() {
       },
     );
 
+    test(
+      ".fetchCoverage() throws an ArgumentError if the given build is null",
+      () {
+        final result = adapter.fetchCoverage(null);
+
+        expect(result, throwsArgumentError);
+      },
+    );
+
+    test(
+      ".fetchCoverage() fetches coverage for the given build",
+      () async {
+        whenFetchCoverage(withWorkflowRun: defaultWorkflowRun)
+            .thenSuccessWith(defaultArtifactsPage);
+
+        final result = await adapter.fetchCoverage(defaultBuild);
+
+        expect(result, isNotNull);
```

Comment:
Let's be more strict here as there should be a concrete value instead of `isNotNull`.